### ISSUE TITLE
[LSP0 and LSP9] Add two step process for renounceOwnership

### ISSUE THREAD

**Issue by frozeman** [lukso-network/LIPs#115]
This creates additional safety for a very dangerous action. Options are:

Call `renounceOwnership()` 2x within 50 blocks

Call `renounceOwnership()` and after `confirmRenounceOwnership()/renounceOwnershipConfirm()` within 50 blocks

**Comment by frozeman** [comment]
Lets go with `Call renounceOwnership()` 2x within 50 blocks

**Comment by frozeman** [comment]
@JeneaVranceanu @YamenMerhi

**Comment by JeneaVranceanu** [comment]
I suppose that instead of me @CJ42 supposed to be tagged :)

**Comment by CJ42** [comment]
`event RenounceOwnershipInitiated(...)`

There should be as well a deadline for renouncing ownership.

```
// renounceOwnership(...).       // ---------------------> timeline to confirm <---------------//                                
block n -----------------> block n + 50 ---------------------------------------------> block n + 150 ------------------> timeout
```

After `block n + 150`, this is a timeout.

**Comment by YamenMerhi** [comment]
It was decided to make `renounceOwnership(..)`   a two step process action, where the the user calls `renounceOwnership(..)`, which starts a time in blocks, where for the first 100 blocks, he can not complete the renouncement, and after 100 blocks he has the ability to call `renounceOwnership(..)` again to complete the renouncement. If after 200 blocks after the initial call to renounceOwnership() the process was not completed, the user will have to start all over again. We also added a **RenounceOwnershipInitiated** Event to notify the user that `renounceOwnership(..)` was called for the first time.

This will allow for a delay time for the owner to act, if a malicious controller initiated the process. And allows for an expiry, the process is not completed within 100 blocks (~20min)

**Comment by b00ste** [comment]
@frozeman Can we use `block.timestamp` instead of `block.number`?

**Comment by CJ42** [comment]
Closing as implemented in #129 ✅
